app/main/views/suppliers.py

```diff
@@ -22,9 +22,13 @@ def list_suppliers():
     suppliers = Supplier.query.order_by(Supplier.name)
 
     if prefix:
-        # case insensitive LIKE comparison for matching supplier names
-        suppliers = suppliers.filter(
-            Supplier.name.ilike(prefix + '%'))
+        if prefix == 'other':
+            suppliers = suppliers.filter(
+                Supplier.name.op('~')('^[^A-z]+'))
```

Comment:
`[A-z]` includes other ascii characters, this should be `[A-Za-z]`.
Also, we are only interested in the first character so the `+` could be removed.